### PR TITLE
[KOTS]: add recommended preflights

### LINKED ISSUE #8846: Add memory check
See [docs](https://troubleshoot.sh/docs/analyze/node-resources/#available-filters)

### LINKED ISSUE #8856: Add a check for a default StorageClass
See [docs](https://troubleshoot.sh/docs/analyze/storage-class)

### PR DESCRIPTION
**Pull request by MrSimonEmms**
## Description

This is derived from #8840 and the initial work on the recommended items

## Related Issue(s)

Fixes #8846 - memory check
Fixes #8856 - default storage class

## How to test

Deploy via KOTS and run preflights

## Release Notes

```release-note
[KOTS]: add recommended preflights
```

### CODE REVIEW

**Review by nandajavarma — COMMENTED** [review]
The changes look good to me. I was trying to test it out with the image built with this PR, but I still see the old preflights. I might be missing something. will get back.

**Comment by MrSimonEmms** [comment]
> The changes look good to me. I was trying to test it out with the image built with this PR, but I still see the old preflights. I might be missing something. will get back.

It's not in the `unstable` channel. You'll need to run `make` and publish to your own dev channel

**Comment by MrSimonEmms** [comment]
@Pothulapati we do actually create PVCs for in cluster Docker registry and Minio

**Review by Pothulapati — APPROVED** [review]
Changes LGTM then!

File: install/kots/manifests/kots-preflight.yaml [hunk 1]
```diff
@@ -84,8 +84,16 @@ spec:
               message: There must be at least one node with the label "gitpod.io/workload_workspace_headless" and value "true"
           - pass:
               message: Node with label exists
+    - customResourceDefinition:
```
Comment (MrSimonEmms) [inline install/kots/manifests/kots-preflight.yaml:87]: I moved this up so the memory/CPU checks are in the same section - seemed more logical that way

**Review by Pothulapati — APPROVED** [review]
:shipit:

**Pull request merged**